microscopic-monks/primal/engine/perlin.py

```diff
@@ -0,0 +1,66 @@
+from math import floor
```

Comment:
I do like that you change the old version that used NumPy with this one. 
it uses very smart bitwise operations, it also does not use magic numbers, that is great to see. 
Functions are short, clear and easy to read. The private functions are correctly marked and it's much better structured than the previous version.